app/test_main.py

```diff
@@ -1 +1,28 @@
-# write your code here
+import pytest
+from unittest import mock
+
+from app.main import can_access_google_page
+
+
+@pytest.fixture()
+def mocked_google_url():
+    with mock.patch("app.main.valid_google_url") as mock_url:
+        yield mock_url
+
+
+@pytest.fixture()
+def mocked_internet_connection():
+    with mock.patch("app.main.has_internet_connection") as mock_connection:
+        yield mock_connection
+
+
+def test_google_page_true(mocked_google_url, mocked_internet_connection):
+    mocked_google_url.return_value = True
+    mocked_internet_connection.return_value = True
+    assert can_access_google_page(100) == "Accessible"
+
+
+def test_google_page_false(mocked_google_url, mocked_internet_connection):
+    mocked_google_url.return_value = False
+    mocked_internet_connection.return_value = True
+    assert can_access_google_page(400) == "Not accessible"
```

Comment:
i think u must check all variants of True and False
True , False
False, False
And why are u using int in "can_access_google_page(400)" it's doesn't matter but little confused

Reply:
I agree with Igor, in tests we should cover all possible variants, it works now with your solution, but what if `can_access_google_page` was written with mistakes?